### PR TITLE
Fix and re-enable disabled unit test suites

### LINKED ISSUE #7016: Timers don't count down or up to target date properly
#### Summary

When creating or editing a Timer, users can set a target date for the timer to count down or up to. This is broken, and on pressing play the timer always just resets to 0 and counts up from there.

#### Expected vs Current Behavior

The timer should count down or up to the specified target date

#### Steps to Reproduce

1. Create a Timer
2. Specify a target date
3. Press play
4. Note that the timer does not start counting until you reset from 0

#### Environment

* Open MCT Version:  3.0.1
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### LINKED ISSUE #7019: `hideTree` and `hideInspector` url param no longer hides the tree and inspector
#### Summary

URLs with `hideTree=true`and/or `hideInspector=true` in them used to hide the left pane tree or inspector. 

@unlikelyzero edit: This has been broken for quite some time.

#### Expected vs Current Behavior

It should hide and inspector the tree on load.

#### Steps to Reproduce

1. Add `hideTree=true` as a url param to any openmct url and navigate to it
2. Notice that the tree is still expanded

#### Environment

* Open MCT Version:  >=3.0.0
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by ozyx**
Closes #7016
Closes #7019  

### Describe your changes:

Fixes and re-enables most unit test suites that were disabled while we were working through the Vue 3 compat upgrade. Fixes a couple of issues and adds e2e coverage for suites we were unable to re-enable.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/ui/layout/pane.vue [hunk 1]
```diff
@@ -104,6 +104,10 @@ export default {
     this.type = this.$parent.type;
     this.styleProp = this.type === 'horizontal' ? 'width' : 'height';
   },
+  created() {
+    // Hide tree and/or inspector pane if specified in URL
+    this.openmct.router.on('change:params', this.handleHideUrl.bind(this));
```
Comment (scottbell) [inline src/ui/layout/pane.vue:109]: clean!

File: src/plugins/timer/components/Timer.vue [hunk 2]
```diff
@@ -78,7 +78,11 @@ export default {
       }
     },
     startTimeMs() {
-      return Date.parse(this.configuration.timestamp);
+      if (Number.isNaN(Date.parse(this.configuration.timestamp))) {
```
Comment (scottbell) [inline src/plugins/timer/components/Timer.vue:81]: Should we add a comment here why the timestamp might not be a number?

**Review by scottbell — CHANGES_REQUESTED** [review]
Good stuff @ozyx! Changes work well:


https://github.com/nasa/openmct/assets/9853862/1d76f158-f128-4af3-90a2-fd649720f954


https://github.com/nasa/openmct/assets/9853862/fab9d97e-3575-4d4d-aa7d-a068bbb51b81

Had a small suggestion about code comments, but otherwise looks good.

**Comment by ozyx** [comment]
> Good stuff @ozyx! Changes work well:
>  timers.mov 
> Had a small suggestion about code comments, but otherwise looks good.

Hmm.. in your Timer video, that behavior doesn't seem quite right. You can see in the inspector the Target is initially undefined? Then when we hit play, it resets the target completely. Will look into this as well.

File: e2e/tests/functional/plugins/timer/timer.e2e.spec.js [hunk 3]
```diff
@@ -63,6 +66,70 @@ test.describe('Timer', () => {
...
+  let timer;
+
+  test.beforeEach(async ({ page }) => {
+    await page.goto('./', { waitUntil: 'domcontentloaded' });
+    timer = await createDomainObjectWithDefaults(page, { type: 'timer' });
+    await assertTimerElements(page, timer);
+  });
+
+  // Override clock
+  test.use({
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/timer/timer.e2e.spec.js:79]: Let's document this in our e2e best practices

File: src/plugins/timer/components/Timer.vue [hunk 4]
```diff
@@ -78,7 +78,11 @@ export default {
...
+        return this.configuration.timestamp;
+      } else {
+        return Date.parse(this.configuration.timestamp);
```
Comment (unlikelyzero) [inline src/plugins/timer/components/Timer.vue:84]: isn't this covered by your timer.e2e.spec.js?

File: e2e/tests/functional/plugins/timer/timer.e2e.spec.js [hunk 5]
```diff
@@ -157,3 +227,23 @@ async function assertTimerStateAfterAction(page, action) {
...
   await expect.soft(page.locator('.c-timer')).toHaveClass(new RegExp(timerStateClass));
 }
+
+/**
+ * Assert that all the major components of a timer are present in the DOM.
+ * @param {import('@playwright/test').Page} page
+ * @param {import('../../../../appActions').CreatedObjectInfo} timer
+ */
+async function assertTimerElements(page, timer) {
+  const timerElement = page.locator('.c-timer');
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/timer/timer.e2e.spec.js:237]: let's add the a11y property to this

File: e2e/appActions.js [hunk 6]
```diff
@@ -78,7 +78,7 @@ async function createDomainObjectWithDefaults(
 
   // Navigate to the parent object. This is necessary to create the object
   // in the correct location, such as a folder, layout, or plot.
-  await page.goto(`${parentUrl}?hideTree=true`);
+  await page.goto(`${parentUrl}`);
```
[the remark this answers is not in the document]
Comment (ozyx) [reply e2e/appActions.js:81]: it shouldn't be here and it was only working because hideTree was broken

File: e2e/README.md [hunk 7]
```diff
@@ -381,6 +381,27 @@ For best practices with regards to mocking network responses, see our [couchdb.e
 
 The following contains a list of tips and tricks which don't exactly fit into a FAQ or Best Practices doc.
 
+- Overriding the Clock
```
Comment (unlikelyzero) [inline e2e/README.md:384]: Overriding the Browser's Clock to control time based elements

File: e2e/appActions.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (unlikelyzero) [reply e2e/appActions.js:81]: @ozyx i remember now: we put that in there to make the e2e tests faster when running on viper. If we launch with the tree closed, we load faster

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
We need to add hideTree back to improve the speed of execution when using appActions in viper

File: e2e/appActions.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (ozyx) [reply e2e/appActions.js:81]: I don't think so, this never worked after the initial navigation. My changes make it work like a real router param. Maybe we can include it somewhere else?

**Review by unlikelyzero — APPROVED** [review]

**Review by scottbell — APPROVED** [review]
Timers working well (from what I can tell!):

https://github.com/nasa/openmct/assets/9853862/55b0f8df-961a-44f0-bc94-efd06f5297b8

**Pull request merged**